File: admin/src/main/java/com/project/admin/controller/dto/response/techblog/TechBlogDetailResponse.java
package com.project.admin.controller.dto.response.techblog;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.project.admin.constant.Region;
import com.project.admin.constant.Source;
import com.project.admin.constant.Status;
import com.project.admin.service.dto.techblog.TechBlogDetailResult;
import java.time.LocalDateTime;
import java.util.List;

public record TechBlogDetailResponse(
        String id,
        Source source,
        Region region,
        String title,
        String url,
        @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss", timezone = "Asia/Seoul")
        LocalDateTime publishedAt,
        String publisher,
        Status status,
        String summaries,
        List<String> tags
) {

    public static TechBlogDetailResponse of(TechBlogDetailResult result) {
        return new TechBlogDetailResponse(
                String.valueOf(result.id()),
                result.source(),
                result.region(),
                result.title(),
                result.url(),
                result.publishedAt(),
                result.publisher(),
                result.status(),
                result.summaries(),
                result.tags()
        );
    }
}

File: api/src/main/kotlin/com/project/api/controller/dto/TechBlogDetailResponse.kt
package com.project.api.controller.dto

import com.project.api.constants.Region
import com.project.api.constants.Source
import com.project.api.service.dto.TechBlogDetailResult
import java.time.LocalDateTime

data class TechBlogDetailResponse(
    val id: String,
    val source: Source,
    val region: Region,
    val title: String,
    val url: String,
    val publishedAt: LocalDateTime,
    val tags: List<String>,
    val summary: String?,
) {
    companion object {
        fun of(result: TechBlogDetailResult) = TechBlogDetailResponse(
            id = result.id.toString(),
            source = result.source,
            region = result.region,
            title = result.title,
            url = result.url,
            publishedAt = result.publishedAt,
            tags = result.tags,
            summary = result.summary,
        )
    }
}

